python/pyproject.toml

```diff
@@ -55,6 +55,8 @@ pyyaml = "6.0.0"
 pydantic = "1.10.2"
 fsspec = "2022.10.0"
 
+typing-extensions = 'typing-extensions==4.4.0'
```

Comment:
Is this optional?

Reply:
Only if you use Python 3.11 (which we don't yet officially support).

Reply:
Moved `typing-extensions` to the dev dependencies